flake.nix

```diff
@@ -79,14 +79,96 @@
     {
       packages = forAllSystems (system:
         let
+          inherit (pkgs) lib;
           pkgs = nixpkgs.legacyPackages.${system};
           options = mkDocOptions pkgs;
+          filterOptions = import ./filterOptions.nix lib;
+          evaluatedModules = pkgs.lib.evalModules {
+            modules = [
+              ./src/modules/top-level.nix
+            ];
+            specialArgs = { inherit pre-commit-hooks pkgs inputs; };
+          };
+          generate-key-options = key:
+            filterOptions
+              (path: option:
+                lib.any (lib.hasSuffix "/${key}.nix") option.declarations)
+              evaluatedModules.options;
+
+          optionsDocs = optionParameter: pkgs.nixosOptionsDoc {
+            options = optionParameter;
+            variablelistId = "options";
+            transformOptions = options: removeAttrs options [ "declarations" ];
+          };
         in
         {
           default = self.packages.${system}.devenv;
           devenv = mkPackage pkgs;
           devenv-docs-options = options.optionsCommonMark;
           devenv-docs-options-json = options.optionsJSON;
+          devenv-generate-individual-docs =
+            let
+              inherit (pkgs) lib;
+              languageOptions = builtins.mapAttrs (key: _: generate-key-options key) evaluatedModules.config.languages;
+              serviceOptions = builtins.mapAttrs (key: _: generate-key-options key) evaluatedModules.config.services;
+              processManagersOptions = builtins.mapAttrs (key: _: generate-key-options key) evaluatedModules.config.process-managers;
+              processedOptions = option: builtins.mapAttrs (key: options: optionsDocs options) option;
+            in
+            pkgs.stdenv.mkDerivation {
+              name = "generate-individual-docs";
+              src = ./.;
+              buildPhase = ''
```

Comment:
Not really important, just suggestions:
1. `runCommand` is a common helper for derivations like these.
2. If you don't need the entire source, you can set `src = ./docs` to avoid copying the entire source to the store.

Reply:
I tried using `runCommand` and I don't think it handles the `src` attributes gracefully:

https://discourse.nixos.org/t/using-the-src-attribute-in-runcommand/19329